From 484ff39cf20a72a0ae3598936ffc921c16409d07 Mon Sep 17 00:00:00 2001
From: "Prakash, Divya1" <divya1.prakash@intel.com>
Date: Wed, 30 May 2018 00:38:39 +0530
Subject: [PATCH] ASOC: Intel: Skylake: Reset pipe in POST_PMD DAPM

Reset pipe in POST_PMD DAPM  event as skl_pcm_hw_free()
is not called in suspend-resume scenario.

Signed-off-by: Prakash, Divya1 <divya1.prakash@intel.com>
Signed-off-by: Tan Seng Kai <seng.kai.tan@intel.com>
---
 sound/soc/intel/skylake/skl-topology.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sound/soc/intel/skylake/skl-topology.c b/sound/soc/intel/skylake/skl-topology.c
index fabf1df..f7106a0 100644
--- a/sound/soc/intel/skylake/skl-topology.c
+++ b/sound/soc/intel/skylake/skl-topology.c
@@ -1871,6 +1871,7 @@ static int skl_tplg_mixer_dapm_post_pmd_event(struct snd_soc_dapm_widget *w,
 		src_module = dst_module;
 	}
 
+	skl_reset_pipe(ctx, mconfig->pipe);
 	skl_delete_pipe(ctx, mconfig->pipe);
 
 	list_for_each_entry(w_module, &s_pipe->w_list, node) {
-- 
1.9.1

